lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1524,7 +1532,7 @@ void TNonblockingIOThread::breakLoop(bool error) {
   }
 
   // sets a flag so that the loop exits on the next event
-  event_base_loopbreak(eventBase_);
+  event_base_loopexit(eventBase_, 0);
```

Comment:
I created the JIRA entry, with a patch that works for us (in production and testing). I know it is not the ideal fix, but I was hoping that someone who is more versed in the ``TNonblockingServer`` implementation can work it. I don't have the time to dig into the design of ``TNonblockingServer``.